### PR TITLE
Citizen - Happiness - /Bug fix #2799

### LINKED ISSUE #2799: Citizen - Happiness - /Bug fix
On the citizen windows,  there is now a "next" and "previous" button on the gui.  This will allow you to toggle between the main view and the happiness view.

The Happiness view allows you to view the different modifiers that affect that citizens happiness.   Each value has a color to it.  
"RED" means has a negative affect on their happiness
"YELLOW" means that it has no affect on their happiness
"GREEN" means it has a positive affect on their happiness

Also, BUG.   The no fields modifier was not checking to make sure they are a "FARMER" and was applying it to every citizen.  That is why most citizen couldn't get above a value of 6.0

This window shows all modifiers and some of the modifiers will not apply to that citizen.  Those values will be always "YELLOW" meaning no affect.

Having everything "YELLOW" will get you a value of 8.  to get higher you need some in green.

### PR DESCRIPTION
**Pull request by bob0bob**
On the citizen windows, there is now a "next" and "previous" button on the gui. This will allow you to toggle between the main view and the happiness view.

The Happiness view allows you to view the different modifiers that affect that citizens happiness. Each value has a color to it.
"RED" means has a negative affect on their happiness
"YELLOW" means that it has no affect on their happiness
"GREEN" means it has a positive affect on their happiness

Also, BUG. The no fields modifier was not checking to make sure they are a "FARMER" and was applying it to every citizen. That is why most citizen couldn't get above a value of 6.0

This window shows all modifiers and some of the modifiers will not apply to that citizen. Those values will be always "YELLOW" meaning no affect.

Having everything "YELLOW" will get you a value of 8. to get higher you need some in green.

Closes #
Closes #
Closes #

# Changes proposed in this pull request:
-
-
-

Review please

### CODE REVIEW

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 40183 |  1  |  1  |  0   |
| Errors |  597  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=16347&tab=Inspection&buildTypeId=bt88)

File: src/api/java/com/minecolonies/api/util/constant/WindowConstants.java [hunk 1]
```diff
@@ -696,6 +696,16 @@
     public static final String LABEL_CONSTRUCTION_NAME =  "constructionName";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/WindowConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name LABEL_CONSTRUCTION_NAME doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/WindowConstants.java [hunk 2]
```diff
@@ -696,6 +696,16 @@
...
     public static final String LABEL_CONSTRUCTION_POS =  "constructionPos";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/WindowConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name GUI_BUTTON_RECALCULATE doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/WindowConstants.java [hunk 3]
```diff
@@ -981,6 +991,21 @@
...
+
+    /**
+     * Button leading the player to the previous page.
+     */
+    public static final String BUTTON_NEXT_PAGE = "nextPage";
+
+    /**
+     * ID for happiness modifier view
+     */
+    public static final String HAPPINESS_MODIFIER_PANE = "happinessModifierView";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/WindowConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name HAPPINESS_MODIFIER_PANE doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/WindowConstants.java [hunk 4]
```diff
@@ -981,6 +991,21 @@
      */
     public static final int LIFE_COUNT_DIVIDER = 30;
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/WindowConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name LIFE_COUNT_DIVIDER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/WindowConstants.java [hunk 5]
```diff
@@ -696,6 +696,16 @@
     public static final String LABEL_CONSTRUCTION_NAME =  "constructionName";
     public static final String LABEL_CONSTRUCTION_POS =  "constructionPos";
 
+    /**
+     * The fields used into the happiness
+     */
+    public static final String FOOD_MODIFIER    = "foodLevel";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/WindowConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name FOOD_MODIFIER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/WindowConstants.java [hunk 6]
```diff
@@ -696,6 +696,16 @@
...
+    public static final String HOUSE_MODIFIER   = "houseLevel";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/WindowConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name HOUSE_MODIFIER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/WindowConstants.java [hunk 7]
```diff
@@ -981,6 +991,21 @@
      */
     public static final int LIFE_COUNT_DIVIDER = 30;
 
+    /**
+     * Button leading the player to the next page.
+     */
+    public static final String BUTTON_PREV_PAGE = "prevPage";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/WindowConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name BUTTON_PREV_PAGE doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/WindowConstants.java [hunk 8]
```diff
@@ -696,6 +696,16 @@
     public static final String LABEL_CONSTRUCTION_NAME =  "constructionName";
     public static final String LABEL_CONSTRUCTION_POS =  "constructionPos";
 
+    /**
+     * The fields used into the happiness
+     */
+    public static final String FOOD_MODIFIER    = "foodLevel";
+    public static final String HOUSE_MODIFIER   = "houseLevel";
+    public static final String DAMAGE_MODIFIER  = "damageLevel";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/WindowConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name DAMAGE_MODIFIER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/WindowConstants.java [hunk 9]
```diff
@@ -696,6 +696,16 @@
...
+    public static final String JOB_MODIFIER     = "jobLevel";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/WindowConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name JOB_MODIFIER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/WindowConstants.java [hunk 10]
```diff
@@ -696,6 +696,16 @@
...
+    public static final String FIELDS_MODIFIER  = "farmsLevel";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/WindowConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name FIELDS_MODIFIER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/WindowConstants.java [hunk 11]
```diff
@@ -696,6 +696,16 @@
...
+    public static final String TOOLS_MODIFIER   = "toolsLevel";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/WindowConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TOOLS_MODIFIER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/WindowConstants.java [hunk 12]
```diff
@@ -981,6 +991,21 @@
...
 
+    /**
+     * Button leading the player to the next page.
+     */
+    public static final String BUTTON_PREV_PAGE = "prevPage";
+
+    /**
+     * Button leading the player to the previous page.
+     */
+    public static final String BUTTON_NEXT_PAGE = "nextPage";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/WindowConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name BUTTON_NEXT_PAGE doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java [hunk 13]
```diff
@@ -75,6 +75,16 @@
      */
     private       int             lifeCount  = 0;
 
+    /**
+     * Button leading to the previous page.
+     */
+    private Button buttonPrevPage;
```
Comment (LDTTeam-Buildsystem) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java]: ### Warnings:
- Instance field name buttonPrevPage doesn't match regex 'm_[a-z][A-Za-z\d]*'

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 40182 |  2  |  3  |  -1  |
| Errors |  597  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=16371&tab=Inspection&buildTypeId=bt88)

File: src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java [hunk 14]
```diff
@@ -75,6 +75,16 @@
...
+
+    /**
+     * Button leading to the next page.
+     */
+    private Button buttonNextPage;
```
Comment (LDTTeam-Buildsystem) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java]: ### Warnings:
- Instance field name buttonNextPage doesn't match regex 'm_[a-z][A-Za-z\d]*'

File: src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java [hunk 15]
```diff
@@ -58,7 +58,7 @@
     /**
      * Scrollinglist of the resources.
      */
-    private final ScrollingList   resourceList;
+    private ScrollingList   resourceList;
```
Comment (LDTTeam-Buildsystem) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java]: ### Warnings:
- Instance field name resourceList doesn't match regex 'm_[a-z][A-Za-z\d]*'

File: src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java [hunk 16]
```diff
@@ -75,6 +75,16 @@
      */
     private       int             lifeCount  = 0;
```
Comment (LDTTeam-Buildsystem) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java]: ### Warnings:
- Instance field name lifeCount doesn't match regex 'm_[a-z][A-Za-z\d]*'

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 40182 |  2  |  3  |  -1  |
| Errors |  597  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=16371&tab=Inspection&buildTypeId=bt88)

File: src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java [hunk 17]
```diff
@@ -275,6 +275,20 @@
     public static final String ON_STRING                                                           = "com.minecolonies.coremod.gui.townHall.on";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name ON_STRING doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java [hunk 18]
```diff
@@ -275,6 +275,20 @@
...
+    @NonNls
+    public static final String CMCG_HAPPINESS_DAMAGE                                               = "com.minecolonies.coremod.gui.happiness.damage";
+    @NonNls
+    public static final String CMCG_HAPPINESS_HOUSE                                                = "com.minecolonies.coremod.gui.happiness.house";
+    @NonNls
+    public static final String CMCG_HAPPINESS_JOB                                                  = "com.minecolonies.coremod.gui.happiness.job";
+    @NonNls
+    public static final String CMCG_HAPPINESS_FARMS                                                = "com.minecolonies.coremod.gui.happiness.farms";
+    @NonNls
+    public static final String CMCG_HAPPINESS_TOOLS                                                = "com.minecolonies.coremod.gui.happiness.tools";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name CMCG_HAPPINESS_TOOLS doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java [hunk 19]
```diff
@@ -275,6 +275,20 @@
     public static final String ON_STRING                                                           = "com.minecolonies.coremod.gui.townHall.on";
     @NonNls
     public static final String OFF_STRING                                                          = "com.minecolonies.coremod.gui.townHall.off";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name OFF_STRING doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java [hunk 20]
```diff
@@ -275,6 +275,20 @@
...
+    @NonNls
+    public static final String CMCG_HAPPINESS_HAPPINESSMODIFIER                                    = "com.minecolonies.coremod.gui.happiness.happinessModifier";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name CMCG_HAPPINESS_HAPPINESSMODIFIER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java [hunk 21]
```diff
@@ -275,6 +275,20 @@
...
+    @NonNls
+    public static final String CMCG_HAPPINESS_FOOD                                                 = "com.minecolonies.coremod.gui.happiness.food";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name CMCG_HAPPINESS_FOOD doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java [hunk 22]
```diff
@@ -275,6 +275,20 @@
...
+    @NonNls
+    public static final String CMCG_HAPPINESS_DAMAGE                                               = "com.minecolonies.coremod.gui.happiness.damage";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name CMCG_HAPPINESS_DAMAGE doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java [hunk 23]
```diff
@@ -275,6 +275,20 @@
...
+    @NonNls
+    public static final String CMCG_HAPPINESS_HOUSE                                                = "com.minecolonies.coremod.gui.happiness.house";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name CMCG_HAPPINESS_HOUSE doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 40199 | 18  |  1  | +17  |
| Errors |  597  |  0  |  0  |  +0  |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=16413&tab=Inspection&buildTypeId=bt88)

File: src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java [hunk 24]
```diff
@@ -275,6 +275,20 @@
...
+    @NonNls
+    public static final String CMCG_HAPPINESS_JOB                                                  = "com.minecolonies.coremod.gui.happiness.job";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name CMCG_HAPPINESS_JOB doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java [hunk 25]
```diff
@@ -275,6 +275,20 @@
...
+    @NonNls
+    public static final String CMCG_HAPPINESS_FARMS                                                = "com.minecolonies.coremod.gui.happiness.farms";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name CMCG_HAPPINESS_FARMS doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java [hunk 26]
```diff
@@ -44,7 +44,7 @@
 
 import static com.minecolonies.api.util.constant.Suppression.RAWTYPES;
```
Comment (LDTTeam-Buildsystem) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java]: ### Errors:
- Unused import import static com.minecolonies.api.util.constant.Suppression.RAWTYPES;

File: src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java [hunk 27]
```diff
@@ -44,7 +44,7 @@
...
 import static com.minecolonies.api.util.constant.WindowConstants.*;
```
Comment (LDTTeam-Buildsystem) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java]: ### Errors:
- Unused import import static com.minecolonies.api.util.constant.WindowConstants.*;

File: src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java [hunk 28]
```diff
@@ -44,7 +44,7 @@
...
-
```
Comment (LDTTeam-Buildsystem) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java]: ### Errors:
- Unused import import static com.minecolonies.api.util.constant.TranslationConstants.*;

File: src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java (hunk 16 above)
Comment (LDTTeam-Buildsystem) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java]: ### Warnings:
- Instance field name lifeCount doesn't match regex 'm_[a-z][A-Za-z\d]*'

File: src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java (hunk 13 above)
Comment (LDTTeam-Buildsystem) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java]: ### Warnings:
- Instance field name buttonPrevPage doesn't match regex 'm_[a-z][A-Za-z\d]*'

File: src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java (hunk 15 above)
Comment (LDTTeam-Buildsystem) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java]: ### Warnings:
- Instance field name resourceList doesn't match regex 'm_[a-z][A-Za-z\d]*'

File: src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java (hunk 14 above)
Comment (LDTTeam-Buildsystem) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java]: ### Warnings:
- Instance field name buttonNextPage doesn't match regex 'm_[a-z][A-Za-z\d]*'

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 40199 | 18  |  1  | +17  |
| Errors |  597  |  0  |  0  |  +0  |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=16413&tab=Inspection&buildTypeId=bt88)

**Review by marchermans — APPROVED** [review]

**Pull request merged**